bodhi/tests/server/functional/test_updates.py

```diff
@@ -2520,6 +2520,353 @@ def test_edit_testing_update_reset_karma_with_same_tester(self, publish, *args):
 
     @mock.patch(**mock_valid_requirements)
     @mock.patch('bodhi.server.notifications.publish')
+    def test_has_negative_karma(self, publish, *args):
+        """
+        The test asserts True to has_negative_karma i.e, if update has negative karma
+        """
+        user = User(name=u'bob')
+        self.db.add(user)
+        self.db.flush()
+
+        nvr = 'bodhi-2.1-1.fc17'
+        args = self.get_update(nvr)
+        resp = self.app.post_json('/updates/', args).json_body
+        publish.assert_called_with(topic='update.request.testing', msg=ANY)
+
+        up = self.db.query(Update).filter_by(title=nvr).one()
+        up.request = None
+        up.status = UpdateStatus.testing
+        self.db.flush()
+
+        # The user gives negative karma first
+        up.comment(self.db, u'Failed to work', author=u'luke', karma=-1)
+        up = self.db.query(Update).filter_by(title=nvr).one()
+        self.assertTrue(up.has_negative_karma)
+
+    @mock.patch(**mock_valid_requirements)
+    @mock.patch('bodhi.server.notifications.publish')
+    def test_has_negative_karma_with_changed_karma(self, publish, *args):
+        """
+        The test asserts False to has_negative_karma i.e, if update has no negative karma finally
+        Here an user first gives negative karma and gives positive karma later to the update
```

Comment:
I'd rephrase this "This test asserts that has_negative_karma returns False when a user posts negative karma and then later posts positive karma."